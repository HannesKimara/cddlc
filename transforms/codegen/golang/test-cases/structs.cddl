person = [name: tstr, age: uint]

pii = (
    age: int,
    name: tstr,
    employer: tstr,
)

employee = (
    ? pii: pii
)

dog = {
    0: int, ; age
    1: tstr ; name
    2: float ; leash-length
}

response = (
  request-id: int
)

device-capability = &(
  receive-audio: 1
  receive-video: 2
)

content = {
    url: text
}

device-info = {
  response
  display-name: text
  state: {
    power: bool
    standby: bool
  }
  capabilities: [* device-capability]
  secret: bytes .size 64
  ? now-playing: {content}
}