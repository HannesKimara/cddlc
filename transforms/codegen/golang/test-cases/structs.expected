package lib

// (cddlc) Ident: person
type Person struct {
	_ struct {
	} `cbor:",toarray"`
	Name string `cbor:"name"`
	Age  uint   `cbor:"age"`
}

// (cddlc) Ident: pii
type Pii struct {
	Age      int    `cbor:"age"`
	Name     string `cbor:"name"`
	Employer string `cbor:"employer"`
}

// (cddlc) Ident: employee
type Employee struct {
	Pii *Pii `cbor:"pii,omitempty"`
}

// (cddlc) Ident: dog
type Dog struct {
	IntKey_0 int     `cbor:"0"`
	IntKey_1 string  `cbor:"1"`
	IntKey_2 float64 `cbor:"2"`
}

// (cddlc) Ident: response
type Response struct {
	RequestId int `cbor:"request-id"`
}

// (cddlc) Ident: DeviceCapability
type DeviceCapability uint

const (
	DeviceCapabilityReceiveAudio DeviceCapability = 1
	DeviceCapabilityReceiveVideo DeviceCapability = 2
)

// (cddlc) Ident: content
type Content struct {
	Url string `cbor:"url"`
}

// (cddlc) Ident: device-info
type DeviceInfo struct {
	Response
	DisplayName string `cbor:"display-name"`
	State       struct {
		Power   bool `cbor:"power"`
		Standby bool `cbor:"standby"`
	} `cbor:"state"`
	Capabilities []DeviceCapability `cbor:"capabilities"`
	Secret       []byte             `cbor:"secret"`
	NowPlaying   *struct {
		Content
	} `cbor:"now-playing,omitempty"`
}
